scrapers/catalog_scraper/main.py

```diff
@@ -1,246 +1,133 @@
-            school = school.findNext("p")
-
-            strings = list(school.stripped_strings)
-            school_title = strings[0]
-            school_name_end = school_title.index("(") - 1
-            school_name = school_title[:school_name_end]
-            if school_name not in data:
-                data[school_name] = []
-
-            for dept in strings[1:]:
-                first_space = dept.index(" ")
-                try:
-                    nbsp_idx = dept.index("\u00a0")
-                except ValueError:
-                    nbsp_idx = -1
-                if nbsp_idx > 0 and nbsp_idx < first_space:
-                    first_space = nbsp_idx
-                code = dept[:first_space]
-                name = dept[first_space + 1 :]
-                if code not in departments:
-                    data[school_name].append({"code": code, "name": name})
-                departments.add(code)
-        return data
-
-
-# Gets the data for a year, this is either the school data or the catalog data
-async def get_year(year_data):
-    year, courses_url, schools_url = year_data
-    async with aiohttp.ClientSession() as s:
-        if sys.argv[1] == "catalog":
-            data = {}
-            await scrape_catalog(s, courses_url, data)
-        else:
-            data = await get_schools(s, schools_url)
-            data = list(map(lambda x: {"name": x[0], "depts": x[1]}, data.items()))
-
-        years = year.split("-")
-        for directory in (f"{years[0]}09", f"{years[1]}01"):
-            directory = "data/" + directory
-            os.makedirs(directory, exist_ok=True)
-            with open(f"{directory}/{sys.argv[1]}.json", "w") as outfile:
-                json.dump(data, outfile, sort_keys=False, indent=2)
-
-
-async def get_page_urls(years_data):
-    # await asyncio.gather(*(get_year(year) for year in years_data))
-    for year in tqdm(years_data):
-        await get_year(year)
-
-
-years = asyncio.run(get_years())
-
-if len(sys.argv) == 1:
-    print(f"USAGE: python3 {sys.argv[0]} (catalog|schools)")
-    sys.exit(1)
-
-if sys.argv[-1] == "LATEST_YEAR":
-    print("Parsing single year")
-    asyncio.run(get_page_urls(years[:1]))
-else:
-    asyncio.run(get_page_urls(years))
-
-"""
-# Duplicate the final summer semester to fall of the next year. This is for if
-# the catalog does not update fast enough for the fall semester
-os.makedirs(f"data/{years[0][0].split('-')[1]}09", exist_ok=True)
-try:
-    copyfile(
-        f"data/{years[0][0].split('-')[1]}05/schools.json",
-        f"data/{years[0][0].split('-')[1]}09/schools.json",
-    )
-except:
-    print("schools.json does not exist for the final semester")
+import json
+from lxml import etree
+
+# The api key is public so it does not need to be hidden in a .env file
+BASE_URL = "http://rpi.apis.acalog.com/v1/"
+# It is ok to publish this key because I found it online already public
+DEFAULT_QUERY_PARAMS = "?key=3eef8a28f26fb2bcc514e6f1938929a1f9317628&format=xml"
```

Comment:
Do we need to worry about RPI unauthorizing this key?

Reply:
I mean maybe, but I think that is unlikely. And using the API while it exists is definitely the better method than our non api scraper.